### ISSUE TITLE
Style questions

### ISSUE THREAD

**Issue by OverLordGoldDragon** [kymatio/kymatio#707]
Kymatio deviates afew from PEP8, two things I rather see changed:

 1. Exceeding 80 characters per line: this is the biggest one, as I have editor on left and console on right, half & half, which is just around 83 characters, so any more and I must scroll horizontally. 
 2. Vertical align, input args: stuff like 

```python
compute_params_filterbank(sigma_min, 1, r_psi=r_psi,
                       alpha=alpha)
```
  the `alpha` reads easier when directly under `sigma_min`.

Are maintainers okay with having `scattering1d/` edited this way?

**Comment by eickenberg** [comment]
I'm fully on board with this. That said, we had tentatively agreed on moving to https://black.readthedocs.io/en/stable/ which might follow slightly different conventions. I would be happy with either and would have others weigh in to decide which is used.

The one thing that is important to me is that style changes happen in a separate PR and not entangled with any other contribution.

**Comment by OverLordGoldDragon** [comment]
> not entangled with any other contribution.

Fair enough, will ease review; seems it should be the very last PR before release, then.

**Comment by OverLordGoldDragon** [comment]
Looked around a bit:

 1. I recommend `warnings.warn` -> `logging.warn` for some, as former will show _only once_, so user cannot know whether e.g. new choice of `J` avoids (or reintroduces) [border effects](https://github.com/kymatio/kymatio/blob/dev/kymatio/scattering1d/filter_bank.py#L309). I'd also add more info on resolving, either directly to the body of warning, or via reference (e.g. "see help(module.fn) for info").
     - The [zero division](https://github.com/kymatio/kymatio/blob/dev/kymatio/scattering1d/filter_bank.py#L157) exception could also inform more, e.g. "reduce J or increase input size"
 2. Why so many [checks](https://github.com/kymatio/kymatio/blob/dev/kymatio/scattering1d/frontend/numpy_frontend.py#L26) at runtime? `out_type, average, vectorize` are set and can be caught at instantiation, before even building any filters. Can be done in Base and avoid code duplication.

**Comment by OverLordGoldDragon** [comment]
p.s. below will spare robos of undue labor via `[skip ci]` in commit title (e.g. `added padding [skip ci]`)

```yml
    build:
        if: "! contains(toJSON(github.event.commits.*.message), '[skip ci]')"
```

**Comment by edouardoyallon** [comment]
please propose a pr and ideally incorporate a mechanism to block noobs like me to reject those style guidelines

**Comment by OverLordGoldDragon** [comment]
I recall reading `kymatio` plans to use Black; personally, I already have a [long list](https://github.com/OverLordGoldDragon/ssqueezepy/blob/master/.github/workflows/conda.yml#L56) of disagreements with PEP8, yet Black goes even further _and_ does not allow rule exceptions (except for total line length). I've also never used Black, but it sure sounds noob-proof.

**Comment by edouardoyallon** [comment]
pr pr pr

**Comment by OverLordGoldDragon** [comment]
You were woofing earlier, is this a cat now or do you ask for a pull... cause I can make one, but with `pycodestyle`.

**Comment by edouardoyallon** [comment]
One little PR says more than a long speech! go for the one you like, this feature is missing

**Comment by OverLordGoldDragon** [comment]
I'll see, focused on JTFS for now. I can also just "start" it, that is, have `pycodestyle` tell the hundred ways the code fails, but not fix.

**Comment by edouardoyallon** [comment]
Propose a PR adding this software and I'be happy to assist in correcting the code.
